src/frontend/js/components/DashBoard/Router.spec.tsx

```diff
@@ -0,0 +1,154 @@
+import { render, screen } from '@testing-library/react';
+import * as mockFactories from 'utils/test/factories';
+import { MemoryRouter } from 'react-router-dom';
+import { IntlProvider } from 'react-intl';
+import { QueryClientProvider } from 'react-query';
+import { FonzieUserFactory, PersistedClientFactory, QueryStateFactory } from 'utils/test/factories';
+import { act } from 'react-dom/test-utils';
+import DashBoardRouter from './Router';
+import { REACT_QUERY_SETTINGS, RICHIE_USER_TOKEN } from '../../settings';
+import { SessionProvider } from '../../data/SessionProvider';
+import createQueryClient from '../../utils/react-query/createQueryClient';
+import { location } from '../../utils/indirection/window';
+import { DashBoardRoute } from './routes';
+import ProtectedRoute from './ProtectedRoute';
+
+jest.mock('utils/indirection/window', () => ({
+  location: {
+    assign: jest.fn(),
+  },
+}));
+
+jest.mock('utils/context', () => ({
+  __esModule: true,
+  default: mockFactories
+    .ContextFactory({
+      authentication: { backend: 'fonzie', endpoint: 'https://demo.endpoint' },
+      joanie_backend: { endpoint: 'https://joanie.endpoint' },
+    })
+    .generate(),
+}));
+
+describe('<DashBoardRouter />', () => {
+  const Wrapper = ({
+    initial_entries,
+    routes,
+  }: {
+    initial_entries: Array<string>;
+    routes?: Array<DashBoardRoute>;
+  }) => {
+    if (!routes) {
+      routes = [
+        {
+          path: '/courses',
+          title: 'courses',
+          element: <h2>Courses</h2>,
+        },
+        {
+          path: '/preferences',
+          title: 'preferences',
+          element: <h2>Preferences</h2>,
+        },
+        {
+          path: '/protected_path',
+          title: 'protected',
+          element: (
+            <ProtectedRoute isAllowed={false} redirectPath="/">
+              <h2>Never ever</h2>
+            </ProtectedRoute>
+          ),
+        },
+        {
+          element: <h2>404 Not Found</h2>,
+          path: '*',
+        },
+      ];
+    }
+    return (
+      <QueryClientProvider client={createQueryClient({ persistor: true })}>
+        <IntlProvider locale="en">
+          <SessionProvider>
+            <MemoryRouter initialEntries={initial_entries}>
+              <DashBoardRouter routes={routes} />
+            </MemoryRouter>
+          </SessionProvider>
+        </IntlProvider>
+      </QueryClientProvider>
+    );
+  };
```

Comment:
You could store default routes in variable scoped in the test suite. (Further explanation in comment below).
```suggestion
describe('<DashBoardRouter />', () => {
  const DEFAULT_ROUTES = [
        {
          path: '/courses',
          title: 'courses',
          element: <h2>Courses</h2>,
        },
        {
          path: '/preferences',
          title: 'preferences',
          element: <h2>Preferences</h2>,
        },
        {
          path: '/protected_path',
          title: 'protected',
          element: (
            <ProtectedRoute isAllowed={false} redirectPath="/">
              <h2>Never ever</h2>
            </ProtectedRoute>
          ),
        },
        {
          element: <h2>404 Not Found</h2>,
          path: '*',
        },
      ];
      
  const Wrapper = ({
    initial_entries,
    routes = DEFAULT_ROUTES,
  }: {
    initial_entries: Array<string>;
    routes?: Array<DashBoardRoute>;
  }) => (
      <QueryClientProvider client={createQueryClient({ persistor: true })}>
        <IntlProvider locale="en">
          <SessionProvider>
            <MemoryRouter initialEntries={initial_entries}>
              <DashBoardRouter routes={routes} />
            </MemoryRouter>
          </SessionProvider>
        </IntlProvider>
      </QueryClientProvider>
    );
```